### ISSUE TITLE
Running only legacy wallet or descriptor wallet tests

### ISSUE THREAD

**Issue by kristapsk** [bitcoin/bitcoin#20394]
`test/functional/test_runner.py` has command line arguments to specify legacy wallet or descriptor wallet:
```
  --descriptors         Run test using a descriptor wallet
  --legacy-wallet       Run test using legacy wallets
```

But this is not working. If you will try to run test suite with `--legacy-wallet` argument, you will have errors on all descriptor wallet tests.
```
1/194 - wallet_hd.py --descriptors failed, Duration: 0 s

stdout:


stderr:
usage: wallet_hd.py [options]
wallet_hd.py: error: argument --legacy-wallet: not allowed with argument --descriptors


2/194 - wallet_backup.py --descriptors failed, Duration: 1 s

stdout:


stderr:
usage: wallet_backup.py [options]
wallet_backup.py: error: argument --legacy-wallet: not allowed with argument --descriptors

```
By looking at `--help` output, I was assuming this specifies to run only either descriptor wallet or legacy wallet tests (skip others), but that's not what's happening. It just adds `--descriptors` or `--legacy-wallet` as argument to individual tests. But, as there are tests with `--descriptors` argument explictly specified in test list, you end up with `test.py --descriptors --legacy-wallet` situations.

I think this should be either fixed or these test_runner parameters be removed.

**Comment by jadijadi** [comment]
Hi All. May I work on this? Since I'm thinking about some refactoring here, let me describe my solution on `test/functional/test_runner.py`:

1. Moving the "DESCRIPTORS_TESTS" and "LEGACY_WALLET_TESTS" tests out of the BASE_SCRIPT and having them as separated lists without the trailing --descriptors and --legacy-wallet
2. When creating ALL_SCRIPTS, adding the DESCRIPTORS & LEGACY_WALLET like this:
```
ALL_SCRIPTS = EXTENDED_SCRIPTS + BASE_SCRIPTS + 
          [ test + ' --descriptors' for test in DESCRIPTORS_TESTS ] +
          [ test + ' --legacy-wallet' for test in LEGACY_WALLET_TESTS ]
```

3. If the --descriptors or --legacy_wallet is provided as arg, just testing on DESCRIPTORS_TESTS or LEGACY_WALLET_TESTS lists. 

If the general idea is fine, I will test & send the PR.

**Comment by willcl-ark** [comment]
This can be closed as the options were removed in [`555519d`](https://github.com/bitcoin/bitcoin/commit/555519d082fbe5e047595f06d7f301e441bb7149#diff-4a04bc0b355c780033960e8c261ee9b6d3c452897e1dcd88a15d272512266c76L197-L202)
